pkg/internal/apis/acme/types_issuer.go

```diff
@@ -182,6 +187,23 @@ type ACMEChallengeSolverHTTP01IngressPodSpec struct {
 	Tolerations []corev1.Toleration
 }
 
+type ACMEChallengeSolverHTTP01IngressTemplate struct {
+	// ObjectMeta overrides for the ingress used to solve HTTP01 challenges.
+	// Only the 'labels' and 'annotations' fields may be set.
+	// If labels or annotations overlap with in-built values, the values here
+	// will override the in-built values.
+	// +optional
+	ACMEChallengeSolverHTTP01IngressObjectMeta `json:"metadata,omitempty"`
+}
+
+type ACMEChallengeSolverHTTP01IngressObjectMeta struct {
+	// Annotations that should be added to the created ACME HTTP01 solver ingress.
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Labels that should be added to the created ACME HTTP01 solver ingress.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
```

Comment:
... aaand here too!

Reply:
I noticed there's no `// +optional` comments or `json` tags in this file, so I removed them.